Taskfile.yml

```diff
@@ -254,10 +247,58 @@ tasks:
         <(cd '{{.OUTPUT_DIR}}' && tar cf - {{.CHECKSUM_TAR_BASE_ARGS}} . | md5sum)
         "{{.CHECKSUM_FILE}}"
 
+  download-tar-and-extract:
+    internal: true
+    requires:
+      vars: ["CHECKSUM_FILE", "FILE_BASE_DIR", "FILE_TAR_NAME", "OUTPUT_DIR", "URL_PREFIX"]
+    label: "Download & Extract {{.FILE_TAR_NAME}}"
+    cmds:
+      - "rm -rf '{{.OUTPUT_DIR}}'"
+      - "mkdir -p '{{.OUTPUT_DIR}}'"
+      - >-
+        curl --fail --location --show-error
+        "{{.URL_PREFIX}}/{{.FILE_TAR_NAME}}"
+        -o "{{.OUTPUT_DIR}}/{{.FILE_TAR_NAME}}"
+      - "tar xf '{{.OUTPUT_DIR}}/{{.FILE_TAR_NAME}}' --directory '{{.OUTPUT_DIR}}'"
+      - >-
+        mv "{{.FILE_BASE_DIR}}/"*
```

Comment:
Note this does not move hidden files (names prefixed with `.`). Let me know if that is a concern.

Reply:
we may set the `dotglob` shopt globally.

> https://taskfile.dev/api/#:~:text=%5B%5Dstring-,Specify%20option%20for%20the%20shopt%20builtin.,-Include%E2%80%8B

Reply:
Unfortunately, `dotglob` is [not yet supported](https://github.com/mvdan/sh/issues/1022).

Reply:
I think it's probably more future proof to handle hidden files. One simple option might be to extract inside '{{.OUTPUT_DIR}}-tmp', move the extracted directory to `{{.OUTPUT_DIR}}`, and then delete `{{.OUTPUT_DIR}}-tmp`.